shared_task_code/ecs_task_queue_consumer.py

```diff
@@ -0,0 +1,131 @@
+        ssm_params = event["ssm_params"]
+        private_subnet = "true" == ssm_params[self.PRIVATE_SUBNETS]
+        network_configuration = {
+            "awsvpcConfiguration": {
+                "subnets": ssm_params[self.SUBNETS].split(","),
+                "securityGroups": [ssm_params[self.SECURITY_GROUP]],
+                "assignPublicIp": "DISABLED" if private_subnet else "ENABLED"
+            }
+        }
+        ecs_params = {
+            "cluster": ssm_params[self.CLUSTER],
+            "networkConfiguration": network_configuration,
+            "taskDefinition": ssm_params[self.IMAGE_ID],
+            "launchType": "FARGATE",
+            "overrides": {
+                "containerOverrides": [{
+                    "name": self.task_name,
+                    "environment": [
+                        {
+                            "name": "TASK_INPUT",
+                            "value": task_params['target']
+                        },
+                        {
+                            "name": "MESSAGE_ID",
+                            "value": message_id
+                        },
+                        {
+                            "name": "RESULTS_BUCKET",
+                            "value": ssm_params[self.RESULTS]
+                        },
+                        {
+                            "name": "S3_METADATA",
+                            "value": "scan_end_time="+task_params['scan_end_time'] +
+                            ",address="+task_params['address'] +
+                            ",address_type="+task_params['address_type']
+
+
+                        }]
+                }]
+            }
+        }
+        print(f"Submitting task: {dumps(ecs_params)}")
+        task_response = self.ecs_client.run_task(**ecs_params)
+        print(f"Submitted scanning task: {dumps(task_response)}")
+
+        failures = task_response["failures"]
+        if len(failures) != 0:
+            raise RuntimeError(
+                f"ECS task failed to start {dumps(failures)}")
+
+        def run_scan(self, scan, message_id):
+            # Use this format to have uniformly named files in S3
+            date_string = f"{datetime.now():%Y-%m-%dT%H%M%S%Z}"
+            s3file = f"{message_id}-{date_string}-{self.task_name}.txt"
+
+            results_filename = f"/tmp/{s3file}"
+
+            if self.func_taskcode != None:
+                self.func_taskcode(self.event, scan, message_id, results_filename, '/tmp/')
+
+            subprocess.check_output(f'cd /tmp;tar -czvf "{s3file}.tar.gz" "{s3file}"', shell=True)
+            s3 = boto3.resource("s3", region_name=self.region)
+            s3.meta.client.upload_file(
+                f"/tmp/{s3file}.tar.gz",
+                self.event["ssm_params"][self.RESULTS],
+                f"{s3file}.tar.gz",
+                ExtraArgs={'ServerSideEncryption': "AES256", 'Metadata': scan})
+
+    def start(self, ValidateData=None, TaskCode=None):

```

Comment:
PEP8 convention is snake case for function parameters, should be e.g. validate_data